### DISCUSSION TITLE
Database size is staying the same size even after deletion of bucket objects

### DISCUSSION THREAD

**Discussion by bennik88** [supabase/supabase#14497] [Questions]
## Problem
I'm Uploading files to my bucket and deleting them afterwards but my used space doesn't get smaller after deletion of the files.

## Example
I'm uploading files to my supabase bucket 'avatars'. The files are in 'id/image_name.jpg
And I delete them using SQL:

```sql
delete from storage.objects where bucket_id = 'avatars' and (storage.foldername(name))[1] = 'be153d4d-a3ed-447c-a27f-7ecb933a6efb';
```

In the supabase UI everything looks great the files and folders are gone after deleting them through SQL. But when i check the reports page and Database usage the size seems to be growing when i repeat that. So it seems like the data is not really deleted.

I've checked with:
```sql
select * from storage.objects;
```
But all objects seem to be removed. Does anyone know what could cause this?

**Comment by GaryAustin1** [comment]
That is not deleting the file data.  You must use storage API calls to delete the files.  If you delete the storage.object row you have orphaned the data in s3.

https://github.com/GaryAustin1/supa-file-helper
Has examples of what needs to be done to delete from SQL, which includes calls to the storage-api with http or pg_net extensions.
